### PR TITLE
changes to manage subscription

### PR DESCRIPTION
**Pull request by VineetBala-AOT**
*Description of changes:*
- Added a new table to store the subscription status for a user
- Subscribe is based on the token so that it expires after the user has subscribed
- Unsubscribe option is independent of the token and based on the user id
- Changes make to email template to have the new format and variable for unsubscribe

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

File: met-api/src/met_api/resources/subscription.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            return str(err), HTTPStatus.INTERNAL_SERVER_ERROR
+
+    @staticmethod
+    @cross_origin(origins=allowedorigins())
+    def patch():
+        """Update a existing subscription partially."""
+        try:
+            requestjson = request.get_json()
+            subscription = SubscriptionSchema().load(requestjson, partial=True)
+            print(subscription)
```
Comment (jadmsaadaot) [inline met-api/src/met_api/resources/subscription.py:81]: debug

File: met-api/src/met_api/models/subscription.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        )
+        db.session.add(new_subscription)
+        if session is None:
+            db.session.commit()
+        else:
+            session.flush()
+        return new_subscription
+
+    @classmethod
+    def update(cls, subscription: SubscriptionSchema, session=None) -> Subscription:
```
Comment (jadmsaadaot) [inline met-api/src/met_api/models/subscription.py:58]: Is this duplicate of the below method?

File: met-web/src/components/engagement/view/ActionContext.tsx [hunk 3]
```diff
@@ -43,48 +45,54 @@ export const ActionContext = createContext<EngagementViewContext>({
...
-                        subText: [
-                            {
-                                text: 'Your email has been verified. You will now receive news and updates from us.',
-                            },
-                        ],
-                    },
-                    type: 'update',
-                }),
-            );
+            if (subscriptionStatus == 'subscribe') {
```
Comment (jadmsaadaot) [inline met-web/src/components/engagement/view/ActionContext.tsx:77]: we should store this 'subscribe' in constant file possible under same Folder, SUBSCRIBE = 'subscribe' same for the other

File: met-web/src/components/engagement/view/ActionContext.tsx [hunk 4]
```diff
@@ -43,48 +45,54 @@ export const ActionContext = createContext<EngagementViewContext>({
...
+                const token = scriptionKey;
+                const subscribed_email = await verifyEmailVerification(token);
+                const subscribed = JSON.stringify(subscribed_email);
+                await createSubscription({
+                    email_verification_id: JSON.parse(subscribed).id,
+                    user_id: JSON.parse(subscribed).user_id,
+                    is_subscribed: 'true',
+                });
+                setSubscriptionText('You have successfully confirmed your subscription. Thank you.');
```
Comment (jadmsaadaot) [inline met-web/src/components/engagement/view/ActionContext.tsx:86]: I think it's better if the state here was isSubscribed, just a boolean, this boolean is picked up by the Subscription component and there you decide the text

File: met-web/src/components/engagement/view/widgets/Subscribe/Subscribed/index.tsx [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { ActionProvider } from '../../../ActionContext';
+import Subscription from './Subscription';
+
+export const ManageSubscription = () => {
+    return (
+        <ActionProvider>
```
Comment (jadmsaadaot) [inline met-web/src/components/engagement/view/widgets/Subscribe/Subscribed/index.tsx:7]: I don't think we should be using a unique ActionProvider more than once. If we need state in multiple pages, we should use Redux

**Review by jadmsaadaot — CHANGES_REQUESTED** [review]
Please look into using Redux for needing to access the subscription data in multiple pages instead of Context

File: met-api/src/met_api/models/subscription.py (hunk 2 above)
Comment (VineetBala-AOT) [reply met-api/src/met_api/models/subscription.py]: removed

File: met-web/src/components/engagement/view/ActionContext.tsx (hunk 3 above)
Comment (VineetBala-AOT) [reply met-web/src/components/engagement/view/ActionContext.tsx]: added in a constant file

**Review by saravanpa-aot — APPROVED** [review]

**Review by tiagograf-AOT — APPROVED** [review]

**Review by jadmsaadaot — APPROVED** [review]

**Pull request merged**